### PR TITLE
feat(snapshot): run test to verify snapshot before upload

### PR DESCRIPTION
**Pull request by tomassusovsky-outreach**
## What this PR does / why we need it

add option to run test to verify snapshot by running test on devenv box provisioned to snapshot before uploading it to storage by providing `--verification-script` parameter containing path to test bash script 
(script can be part of devenv-snapshot project that schedules devenv snapshot creation) 

## Jira ID

[XX-XX]

## Notes for your reviewers

WIP 
- currently this version will generate 2 snapshots not re-upload tested on to real storage
- resolve dependency on e2e package

### CODE REVIEW

File: cmd/devenv/snapshot/snapshot.go [hunk 1]
```diff
@@ -59,13 +60,7 @@ var (
 	`
 	helpersExample = `
 		# Create a snapshot
-		devenv snapshot create
```
Comment (tomassusovsky-outreach) [inline cmd/devenv/snapshot/snapshot.go:62]: these commands in help do not exist

File: cmd/devenv/snapshot/snapshot.go [hunk 2]
```diff
@@ -27,6 +27,7 @@ import (
 	dockerclient "github.com/docker/docker/client"
 	"github.com/getoutreach/devenv/cmd/devenv/destroy"
 	"github.com/getoutreach/devenv/cmd/devenv/provision"
+	devenvinstrumentation "github.com/getoutreach/devenv/internal/e2e/devenv"
```
Comment (tomassusovsky-outreach) [inline cmd/devenv/snapshot/snapshot.go:30]: wip: better not import internal e2e package to prod cmd

File: internal/e2e/devenv/devenv.go [hunk 3]
```diff
@@ -137,7 +137,9 @@ func execOSOut(cmd *exec.Cmd) *exec.Cmd {
 //
 // nolint:gocritic,revive // Why: t is first param in test helpers
 func UseSnapshotStorage(t *testing.T, ctx context.Context, b *box.Config) (func(), error) {
-	t.Log("Setting up snapshot storage infrastructure")
+	if t != nil {
```
Comment (tomassusovsky-outreach) [inline internal/e2e/devenv/devenv.go:140]: wip

File: cmd/devenv/snapshot/snapshot.go [hunk 4]
```diff
@@ -509,6 +533,39 @@ func (o *Options) generateSnapshot(ctx context.Context, s3c *s3.Client,
 	}, nil
 }
 
+func (o *Options) testReprovisionedDevenv(ctx context.Context, verificationScript string) error {
```
Comment (tomassusovsky-outreach) [inline cmd/devenv/snapshot/snapshot.go:536]: approach from https://github.com/getoutreach/devenv/blob/main/internal/e2e/e2e_test.go TestCanProvisionSnapshotDevenv applied here to run test on devenv reprovisioned from new snapshot

File: cmd/devenv/snapshot/snapshot.go [hunk 5]
```diff
@@ -138,6 +133,11 @@ func NewCmdSnapshot(log logrus.FieldLogger) *cli.Command {
 						Usage: "Kubernetes version to use when generating snapshots. Has no effect if a devenv already exists",
 						Value: "v1.21.14",
 					},
+					&cli.StringFlag{
+						Name:  "verification-script",
+						Usage: "Path to the bash script used to verify generated snapshot before uploading it",
+						Value: "",
```
Comment (tomassusovsky-outreach) [inline cmd/devenv/snapshot/snapshot.go:139]: make sure script has exec permission

File: internal/e2e/devenv/devenv.go [hunk 6]
```diff
@@ -223,3 +228,134 @@ func UseSnapshotStorage(t *testing.T, ctx context.Context, b *box.Config) (func(
 	err = execOSOut(exec.CommandContext(ctx, "docker", "network", "connect", "kind", minioContainerName, "--ip", ipAddr)).Run()
 	return cleanupFn, errors.Wrap(err, "failed to pull minio image")
 }
+
+// fix ugly UseSnapshotStorage2
+// UseSnapshotStorage creates a docker backed snapshot storage that can be used
+// for testing. Configures the provided box config to use it.
+//
+// nolint:gocritic,revive // Why: t is first param in test helpers
+func UseSnapshotStorage2(ctx context.Context, b *box.Config, log logrus.FieldLogger) (*s3.Client, func(), error) {
```
Comment (tomassusovsky-outreach) [inline internal/e2e/devenv/devenv.go:237]: wip: resolve UseSnapshotStorage, UseSnapshotStorage2

**Comment by jaredallard** [comment]
I believe this can be closed, right @tomassusovsky-outreach ?

**Comment by jaredallard** [comment]
Re-open if needed!

**Pull request closed without merging**